packages/dd-trace/src/config.js

```diff
@@ -163,6 +172,35 @@ class Config {
       process.env.DD_TRACE_EXPERIMENTAL_TRACEPARENT_ENABLED,
       false
     )
+    const DD_TRACE_PROPAGATION_STYLE_INJECT = propagationStyle(
+      options.tracePropagationStyle && options.tracePropagationStyle.inject,
+      process.env.DD_TRACE_PROPAGATION_STYLE_INJECT,
+      // TODO: change this to traceparent,datadog
+      // This can't be done as a semver-minor until tracestate support also exists
+      'datadog'
+    )
+    const DD_TRACE_PROPAGATION_STYLE_EXTRACT = propagationStyle(
+      options.tracePropagationStyle && options.tracePropagationStyle.extract,
+      process.env.DD_TRACE_PROPAGATION_STYLE_EXTRACT,
+      // TODO: change this to traceparent,datadog
+      // This can't be done as a semver-minor until tracestate support also exists
+      'datadog'
+    )
+    const propagationStyleLists = [
+      DD_TRACE_PROPAGATION_STYLE_INJECT,
+      DD_TRACE_PROPAGATION_STYLE_EXTRACT
+    ]
+    if (isTrue(DD_TRACE_TRACEPARENT_ENABLED)) {
```

Comment:
Let's only set these if the new config is not set since they didn't allow controlling the order which could be confusing.

Reply:
I don't see how it would be confusing if we match the ordering it used before unless configured otherwise.

Reply:
What I'm saying is that users should either use the old variables (hard-coded order, only what is set to true is enabled) and the new config (configurable order and independent inject/extract) but not both at the same time. Not only should this not happen, but it would be confusing because if you enable for example this env var, and then add an order with the new env var, you don't know in advance where the traceparent will be in the list.

Reply:
That's how it works currently. If the experimental configs are used it will retain the old ordering. If the new config is used it will completely override it. We could perhaps include a warning if both configs are used?